### ISSUE TITLE
readFile binary mmcif

### ISSUE THREAD

**Issue by AlessioDelConte** [molstar/molstar#391]
Hi,
I've seen that molstar supports reading binary mmcif files. Now I have converted some mmcif files that I have in binary mmcif, but I can't find a way to load them into molstar.
What I am doing is:

```
const data = await this.plugin.builders.data.readFile({
  file: {file: file, kind: 'file', id: UUID.createv4(), name: file.name},

  isBinary: true,
  // Define file name
  label: file.name,
}, {
  state: {isGhost: true}
});
// Define file format
const format = file.type.split('/').pop() as 'pdb' | 'mmcif';
// Parse data
const trajectory = await this.plugin.builders.structure.parseTrajectory(data.data, format);
// Create model
const model = await this.plugin.builders.structure.createModel(trajectory);
// Create structure
const structure = this.structure = await this.plugin.builders.structure.createStructure(model, {
  name: 'model',
  params: {}
});
```
where file (of type File) is the binary mmcif that I retrieve via a service, but I have some errors on runtime, what I am doing wrong?

Before instead of using readFile i was using rawData, but I saw that it does not support binary inputs.

Thanks if you can help me with this

**Comment by dsehnal** [comment]
Could you be more specific about the errors you are getting?

One thing you can try is read the file into ``ArrayBuffer/Uint8Array`` (one or the other) and use ``this.plugin.builders.data.raw`` function.

**Comment by AlessioDelConte** [comment]
I get an error on the parseTrajectory, more specifically:

Error: TypeError: Cannot read properties of null (reading 'slice')
    at Object.<anonymous> (data.js:347:1)
    at step (tslib.es6.js:102:1)
    at Object.next (tslib.es6.js:83:46)
    at fulfilled (tslib.es6.js:73:43)
    at ZoneDelegate.invoke (zone.js:372:1)
    at Object.onInvoke (core.js:28692:1)
    at ZoneDelegate.invoke (zone.js:371:1)
    at Zone.run (zone.js:134:1)
    at zone.js:1276:1
    at ZoneDelegate.invokeTask (zone.js:406:1)

I've also tried loading the data with loadData but I got the same error.
To give more context I've converted an mmcif file to binary mmcif with the py-mmcif library, can this be a problem?

**Comment by dsehnal** [comment]
This works, perhaps it will help find a bug in your code: https://github.com/molstar/molstar/blob/master/src/mol-plugin-state/actions/file.ts#L64

**Comment by AlessioDelConte** [comment]
Well, I've finally traced the bug, but actually it is not from my side. In the function _parseCifBinary_ there is this line

https://github.com/molstar/molstar/blob/c3afabb4b1189d7e1f795c757b9300786e138b08/src/mol-io/reader/cif/binary/parser.ts#L45

that checks the version of the encoding used, but since I convert to binary mmcif with py-mmcif the unpacked object is the following:
```
{
  version: "0.01",
  encoder: "python-mmcif library",
  dataBlocks: [...]
}
```
so, the match returns null, and the abovementioned error is generated.

Now, I will try to modify the py-mmcif library to modify the version string, but I think it should be useful to modify this check in order to accommodate different types of versioning systems.

Thanks!

**Comment by dsehnal** [comment]
I see. We also have an "official" ``ciftools-python`` with BinaryCIF support in works now too (should be available in 1-2 months), so might be able to move to that afterwards.

**Comment by arose** [comment]
@dsehnal I always thought the `version` field was for the BinaryCIF format not for the encoder library.

**Comment by dsehnal** [comment]
> @dsehnal I always thought the `version` field was for the BinaryCIF format not for the encoder library.

@arose yes it's the format version, not the library one. Then again there is just one version for now...

**Comment by AlessioDelConte** [comment]
Yeah thanks for your kind support. 
Anyway, I modified the py-mmcif library version string by hand to 0.3.0 and re-generated the binary mmcif, but it still won't work during the parsing (now at least it starts to parse it). 
There may be something wrong in the conversion, although when I tried to re-convert back from binary to non binary with the py-mmcif library it seem to work just fine. Well, I think that for now I will just use plain mmcif files, and once the "official" python library is available I will look into that.

Thank you and keep up the good work!
